### PR TITLE
🔀 Update upload api and fix asset_like missing record

### PR DESCRIPTION
**Pull request by mckingho**

### CODE REVIEW

File: server/api/assets.js [hunk 1]
```diff
@@ -81,42 +86,26 @@ router.post(
       if (!isTagsValid(tags)) throw new ValidationError('tags length invalid');
 
       // check license
-      if (!license) throw new ValidationError('asset need a license');
-      if (!isLicenseValid(license))
+      if (!license || !licenseId) {
+        throw new ValidationError('asset need a license');
+      }
+      if (!isLicenseValid(licenseId, license))
         throw new ValidationError('license invalid');
```
Comment (nwingt) [inline server/api/assets.js]: wrap if block with `{}`

File: util/assetUtil.js [hunk 2]
```diff
@@ -1,39 +1,49 @@
...
+      likeFingerprint: assetSHA256,
+      likeFootprint: [],
+      likePreviousVersion: null,
+      type: 'ImageObject',
+      uploadDate: nowDateTime,
+    });
 
     // show MetaMask dialog ask user to sign on MetaMask to continue
     if (ethUtil.onSign) ethUtil.onSign();
+    console.log(tags);
```
Comment (nwingt) [inline util/assetUtil.js]: Please remove this console

File: server/api/assets.js [hunk 3]
```diff
@@ -81,42 +86,26 @@ router.post(
...
 
       // check asset
       const { file: asset } = req;
       const hash256 = sha256(asset.buffer);
       const { height, width } = imageSize(asset.buffer);
-      validateImage(asset, assetSHA256);
```
Comment (inDream) [inline server/api/assets.js]: Should check image hash equal

Comment (mckingho) [reply server/api/assets.js]: `likeFingerprint` is the hash from client. It is validated in the function that server calculates again.

Comment (inDream) [reply server/api/assets.js]: As `hash256` already calculated 2 lines above, you could pass it into `validateImage()`.

**Pull request merged**